tests/consensus_tests/auth_tests/test_jwt_validation.py

```diff
@@ -1,623 +1,37 @@
-    covered_endpoints = set(v.rest_endpoint for v in ACTION_ACCESS.values())
-    for endpoint in endpoint_paths:
-        assert (
-            endpoint in covered_endpoints
-        ), f"REST endpoint `{endpoint}` not found in any of the `ACTION_ACCESS` REST endpoints"
-
-
-class MetadataInterceptor(ClientInterceptor):
-    """A test interceptor that injects invocation metadata."""
-
-    def __init__(self, metadata: List[Tuple[str, str]]):
-        self._metadata = metadata
-
-    def intercept(self, method, request_or_iterator, call_details: ClientCallDetails):
-        """Add invocation metadata to request."""
-        new_details = call_details._replace(metadata=self._metadata)
-        return method(request_or_iterator, new_details)
-
-
-def test_all_grpc_endpoints_are_covered():
-    # read grpc services from the reflection server
-    client: grpc_requests.Client = grpc_requests.Client(
-        GRPC_URI, interceptors=[MetadataInterceptor(API_KEY_METADATA)]
-    )
-
-    # check that all endpoints are covered in ACTION_ACCESS
-    covered_endpoints = set(v.grpc_endpoint for v in ACTION_ACCESS.values())
-
-    for service_name in client.service_names:
-        service = client.service(service_name)
-        for method in service.method_names:
-            grpc_endpoint = f"{service_name}/{method}"
-            assert (
-                grpc_endpoint in covered_endpoints
-            ), f"gRPC endpoint `{grpc_endpoint}` not found in ACTION_ACCESS gRPC endpoints"
+COLL_NAME = "jwt_test_collection"
 
 
 @pytest.fixture(scope="module", autouse=True)
-def uris(tmp_path_factory: pytest.TempPathFactory):
-    extra_env = {
-        "QDRANT__SERVICE__API_KEY": SECRET,
-        "QDRANT__SERVICE__JWT_RBAC": "true",
-        "QDRANT__STORAGE__WAL__WAL_CAPACITY_MB": "1",  # to speed up snapshot tests
-    }
-
+def setup(tmp_path_factory: pytest.TempPathFactory):
     tmp_path = tmp_path_factory.mktemp("api_key_instance")
 
-    peer_api_uris, peer_dirs, bootstrap_uri = start_cluster(
-        tmp_path, num_peers=1, port_seed=PORT_SEED, extra_env=extra_env, headers=API_KEY_HEADERS
-    )
-
-    assert REST_URI in peer_api_uris
+    peer_api_uris, peer_dirs, bootstrap_uri = start_jwt_protected_cluster(tmp_path)
 
     fixtures.create_collection(
         REST_URI,
         collection=COLL_NAME,
-        sharding_method="custom",
         headers=API_KEY_HEADERS,
     )
 
-    requests.put(
-        f"{REST_URI}/collections/{COLL_NAME}/shards",
-        json={"shard_key": SHARD_KEY},
-        headers=API_KEY_HEADERS,
-    ).raise_for_status()
-
-    fixtures.upsert_random_points(
-        REST_URI, 100, COLL_NAME, shard_key=SHARD_KEY, headers=API_KEY_HEADERS
-    )
+    fixtures.upsert_random_points(REST_URI, 100, COLL_NAME, headers=API_KEY_HEADERS)
 
     yield peer_api_uris, peer_dirs, bootstrap_uri
 
     fixtures.drop_collection(REST_URI, COLL_NAME, headers=API_KEY_HEADERS)
+    kill_all_processes()
```

Comment:
We need to kill processes even if `drop_collection` throws an exception.

Also, maybe it's worth to put a fixture with `start_jwt_protected_cluster()` + `kill_all_processes()` into `utils.py`? Then this `setup()` could look like this:

```python
from .utils import jwt_setup
@pytest.fixture(scope="module", autouse=True)
def setup(
    jwt_setup # force the execution order
):
    fixtures.create_collection(
        REST_URI,
        collection=COLL_NAME,
        headers=API_KEY_HEADERS,
    )

    fixtures.upsert_random_points(REST_URI, 100, COLL_NAME, headers=API_KEY_HEADERS)
    yield
    fixtures.drop_collection(REST_URI, COLL_NAME, headers=API_KEY_HEADERS)
    # processes are killed inside jwt_setup fixture
```

Reply:
I had tried something similar, which didn't work because of temp files. But now I've found a better and cleaner solution with your suggestion :)